src/libretro/net/mp.cpp

```diff
@@ -0,0 +1,100 @@
+#include "mp.hpp"
+#include "environment.hpp"
+#include <ctime>
+#include <libretro.h>
+#include <retro_assert.h>
+#include <retro_endianness.h>
+using namespace MelonDsDs;
+
+constexpr long RECV_TIMEOUT_MS = 25;
+
+uint64_t swapToNetwork(uint64_t n) {
+    return swap_if_little64(n);
+}
+
+Packet Packet::parsePk(const void *buf, uint64_t len) {
+    // Necessary because arithmetic on void* is forbidden
+    const char *indexableBuf = (const char *)buf;
+    const char *data = indexableBuf + HeaderSize;
+    retro_assert(len > HeaderSize);
```

Comment:
Looks like this assertion doesn't hold up; I just triggered it in Mario Kart DS. I created the in-game lobby from one of my devices, but as soon as the second device connected this `retro_assert` went off.

Reply:
Try changing it to `len >= HeaderSize`. If it works, then zero sized packets are possible.
Interestingly, I was not able to reproduce this.

Reply:
Managed to reproduce it, but switching to `len >= HeaderSize` fixes it.

Reply:
What game did you test it with? I used Mario Kart DS.

Reply:
Me too. .nds file format, MD5 hash: 8cbf438d5a7c0d9acbe56a9b627ca6c5
The game indeed crashes if the condition is `len > HeaderSize` but switching to `len >= HeaderSize` fixed it for me.

Reply:
I initially tested it with Pokémon HeartGold as well.